Sources/Testing/ExitTests/ExitTest.Result.swift

```diff
@@ -0,0 +1,86 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+#if SWT_NO_EXIT_TESTS
+@available(*, unavailable, message: "Exit tests are not available on this platform.")
+#endif
+extension ExitTest {
+  /// A type representing the result of an exit test after it has exited and
+  /// returned control to the calling test function.
+  ///
+  /// Both ``expect(exitsWith:_:sourceLocation:performing:)`` and
+  /// ``require(exitsWith:_:sourceLocation:performing:)`` return instances of
+  /// this type.
+  public struct Result: Sendable {
+    /// The exit condition the exit test exited with.
+    ///
+    /// When the exit test passes, the value of this property is equal to the
+    /// value of the `expectedExitCondition` argument passed to
+    /// ``expect(exitsWith:_:sourceLocation:performing:)`` or to
+    /// ``require(exitsWith:_:sourceLocation:performing:)``. You can compare two
+    /// instances of ``ExitCondition`` with ``ExitCondition/==(lhs:rhs:)``.
+    public var exitCondition: ExitCondition
+
+    /// Whatever error might have been thrown when trying to invoke the exit
+    /// test that produced this result.
+    ///
+    /// This property is not part of the public interface of the testing
+    /// library.
+    var caughtError: (any Error)?
+
+    @_spi(ForToolsIntegrationOnly)
+    public init(exitCondition: ExitCondition) {
+      self.exitCondition = exitCondition
+    }
```

Comment:
Harnesses that need to handle process spawning themselves instead of using our implementation will need to be able to produce an instance of this type.